handler_ui.go

```diff
@@ -11,25 +11,32 @@ import (
 )
 
 type instrumentTO struct {
-	ID                 uuid.UUID             `json:"id"`
-	Type               InstrumentType        `json:"instrumentType"`
-	Name               string                `json:"name"`
-	ProtocolID         uuid.UUID             `json:"protocolId"`
-	ProtocolName       Protocol              `json:"type"`
-	Enabled            bool                  `json:"enabled"`
-	ConnectionMode     ConnectionMode        `json:"connectionMode"`
-	ResultMode         ResultMode            `json:"runningMode"`
-	CaptureResults     bool                  `json:"captureResults"`
-	CaptureDiagnostics bool                  `json:"captureDiagnostics"`
-	ReplyToQuery       bool                  `json:"replyToQuery"`
-	Status             string                `json:"status"`
-	FileEncoding       string                `json:"fileEncoding"`
-	Timezone           string                `json:"timezone"`
-	Hostname           string                `json:"hostname"`
-	ClientPort         *int                  `json:"clientPort"`
-	AnalyteMappings    []analyteMappingTO    `json:"analyteMappings"`
-	RequestMappings    []requestMappingTO    `json:"requestMappings"`
-	Settings           []instrumentSettingTO `json:"instrumentSettings"`
+	ID                  uuid.UUID             `json:"id"`
+	Type                InstrumentType        `json:"instrumentType"`
+	Name                string                `json:"name"`
+	ProtocolID          uuid.UUID             `json:"protocolId"`
+	ProtocolName        Protocol              `json:"type"`
+	Enabled             bool                  `json:"enabled"`
+	ConnectionMode      ConnectionMode        `json:"connectionMode"`
+	ResultMode          ResultMode            `json:"runningMode"`
+	CaptureResults      bool                  `json:"captureResults"`
+	CaptureDiagnostics  bool                  `json:"captureDiagnostics"`
+	ReplyToQuery        bool                  `json:"replyToQuery"`
+	Status              string                `json:"status"`
+	FileEncoding        string                `json:"fileEncoding"`
+	Timezone            string                `json:"timezone"`
+	Hostname            string                `json:"hostname"`
+	ClientPort          *int                  `json:"clientPort"`
+	FtpUsername         string                `json:"ftpUserName"`
```

Comment:
i'd prefer to keep the FTPConfig enclosed into an object or make these fields nullable, and leave them as nulls if there are no FTPConfig, instead of using empty string

Reply:
changed them to pointers